Annotato/Annotato/Persistence/DocumentsPersistenceManager.swift

```diff
@@ -20,24 +21,57 @@ class DocumentsPersistenceManager {
         self.remoteDocumentsPersistence = RemoteDocumentsPersistence(
             webSocketManager: webSocketManager
         )
+        self.remoteDocumentSharesPersistence = RemoteDocumentSharesPersistence()
 
         setUpSubscribers()
     }
 
     func getOwnDocuments(userId: String) async -> [Document]? {
         let remoteOwnDocuments = await remoteDocumentsPersistence.getOwnDocuments(userId: userId)
-        guard remoteOwnDocuments != nil else {
-            return localDocumentsPersistence.getOwnDocuments(userId: userId)
+        let localOwnDocuments = localDocumentsPersistence.getOwnDocuments(userId: userId)
+
+        if let remoteOwnDocuments = remoteOwnDocuments,
+           let localOwnDocuments = localOwnDocuments {
+            await pruneRemoteDocuments(localDocuments: localOwnDocuments, serverDocuments: remoteOwnDocuments)
```

Comment:
Can we add some comments here to explain what we are doing? Something like 

```swift
// Delete user's own documents on remote when coming from offline to online. 
// User might have deleted documents while offline.
```

But I leave it up to you to change to a better comment or skip this if you feel like its not needed.